extensions/trace_propagators/src/test/java/io/opentelemetry/extensions/trace/propagation/JaegerPropagatorTest.java

```diff
@@ -133,12 +145,49 @@ void inject_NotSampledContext() {
                 TRACE_ID_BASE16, SPAN_ID_BASE16, DEPRECATED_PARENT_SPAN, "0"));
   }
 
+  @Test
+  void inject_SampledContext_withBaggage() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    Context context =
+        Context.current()
+            .with(
+                Span.wrap(
+                    SpanContext.create(
+                        TRACE_ID, SPAN_ID, SAMPLED_TRACE_OPTIONS, TRACE_STATE_DEFAULT)))
+            .with(Baggage.builder().put("foo", "bar").build());
+
+    jaegerPropagator.inject(context, carrier, setter);
+    assertThat(carrier)
+        .containsEntry(
+            PROPAGATION_HEADER,
+            generateTraceIdHeaderValue(
+                TRACE_ID_BASE16, SPAN_ID_BASE16, DEPRECATED_PARENT_SPAN, "1"));
+    assertThat(carrier).containsEntry(BAGGAGE_PREFIX + "foo", "bar");
+  }
+
+  @Test
+  void inject_baggageOnly() {
+    // Metadata won't be propagated, but it MUST NOT cause ay problem.
+    Baggage baggage =
+        Baggage.builder()
+            .put("nometa", "nometa-value")
+            .put("meta", "meta-value", EntryMetadata.create("somemetadata; someother=foo"))
+            .build();
+    Map<String, String> carrier = new LinkedHashMap<>();
+    jaegerPropagator.inject(Context.root().with(baggage), carrier, Map::put);
+    assertThat(carrier)
+        .containsExactlyInAnyOrderEntriesOf(
+            ImmutableMap.of(
+                BAGGAGE_PREFIX + "nometa", "nometa-value",
+                BAGGAGE_PREFIX + "meta", "meta-value"));
+  }
+
   @Test
   void extract_Nothing() {
     // Context remains untouched.
     assertThat(
             jaegerPropagator.extract(
-                Context.current(), Collections.<String, String>emptyMap(), Map::get))
+                Context.current(), Collections.<String, String>emptyMap(), getter))
```

Comment:
might be a little safer/cleaner to just use `Context.root()` here, so we know that we don't have a pre-polluted context.

Reply:
Fair enough, will update.